ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java

```diff
@@ -93,11 +95,60 @@
    *                  {@code false} otherwise
    */
   StoreFindToken(Offset offset, UUID sessionId, UUID incarnationId, boolean inclusive) {
-    this(Type.JournalBased, offset, null, sessionId, incarnationId, inclusive);
+    this(Type.JournalBased, offset, null, sessionId, incarnationId, inclusive, VERSION_2);
   }
 
-  private StoreFindToken(Type type, Offset offset, StoreKey key, UUID sessionId, UUID incarnationId,
-      boolean inclusive) {
+  /**
+   * Uninitialized token with the given version. Refers to the starting of the log.
+   * @param version refers to the version of the token
+   */
+  private StoreFindToken(short version) {
```

Comment:
yes. 3 are public (Unitialized, journal and index). 3 are private ones so that deserialization takes in a version apart from usual args in public constructors. And one private constructor which is called from every other constructor. We can make the deserialization calls to explicitly call the last private with default values, but thought this would mimic how older version tokens are created